src/plugins/discover/public/application/angular/discover.js

```diff
@@ -296,6 +296,14 @@ function discoverController($element, $route, $scope, $timeout, Promise, uiCapab
       appStateContainer,
       data,
       getSavedSearchId: () => savedSearch.id,
+      getSessionName: () => {
+        return (
+          (savedSearch.id && savedSearch.title) ||
+          i18n.translate('discover.discoverDefaultSearchSessionName', {
+            defaultMessage: 'Discover',
+          })
+        );
+      },
```

Comment:
In support of our mission to move code away from `discover.js` do you think you refactor to just use `savedSearch` instead of `getSavedSearchId` and `getSessionName`, since they're both referring to savedSearch? not mandatory. Code LGTM so far, testing follows

Reply:
np, will improve!